okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -941,7 +943,8 @@ public void headers(boolean outFinished,
 
     @Override
     public void rstStream(int streamId, ErrorCode errorCode) {
-      finishStream(streamId, toGrpcStatus(errorCode).augmentDescription("Rst Stream"), null, null);
+      finishStream(
+          streamId, toGrpcStatus(errorCode).augmentDescription("Rst Stream"), false, null, null);
```

Comment:
Done. I restored the original check for CANCELLED or DEADLINE_EXCEEDED status alongside the explicit stopDelivery flag.